src/domain/events/__tests__/EventsSearchPage.test.tsx

```diff
@@ -0,0 +1,169 @@
+import { MockedResponse } from '@apollo/client/testing';
+import { advanceTo, clear } from 'jest-date-mock';
+import React from 'react';
+import wait from 'waait';
+
+import { EventsDocument, Event } from '../../../generated/graphql';
+import {
+  fakeEvents,
+  fakePEvent,
+  fakeLocalizedObject,
+  fakeKeyword,
+  fakeOccurrences,
+  fakeOccurrence,
+} from '../../../utils/mockDataUtils';
+import {
+  render,
+  screen,
+  act,
+  configure,
+  userEvent,
+} from '../../../utils/testUtils';
+import EventsSearchPage from '../EventsSearchPage';
+
+configure({ defaultHidden: true });
+
+const testDate = new Date(2020, 5, 20);
+
+const fakeKeywords = [
+  fakeKeyword({ name: fakeLocalizedObject('vanhukset') }),
+  fakeKeyword({ name: fakeLocalizedObject('nuoret') }),
+  fakeKeyword({ name: fakeLocalizedObject('lapset') }),
+  fakeKeyword({ name: fakeLocalizedObject('ilmainen') }),
+];
+
+const eventMocks: Partial<Event>[] = [
+  {
+    name: fakeLocalizedObject('Testitapahtuma 1'),
+    shortDescription: fakeLocalizedObject('Tapahtuman lyhytkuvaus 1'),
+    pEvent: fakePEvent({
+      nextOccurrence: fakeOccurrences(1, [
+        fakeOccurrence({
+          startTime: new Date(2020, 8, 20, 10, 30).toISOString(),
+        }),
+      ]),
+    }),
+    keywords: fakeKeywords,
+  },
+  {
+    name: fakeLocalizedObject('Testitapahtuma 2'),
+    shortDescription: fakeLocalizedObject('Tapahtuman lyhytkuvaus 2'),
+    pEvent: fakePEvent({
+      nextOccurrence: fakeOccurrences(1, [
+        fakeOccurrence({
+          startTime: new Date(2020, 9, 21, 10, 20).toISOString(),
+        }),
+      ]),
+    }),
+    keywords: fakeKeywords,
+  },
+  {
+    name: fakeLocalizedObject('Testitapahtuma 3'),
+    shortDescription: fakeLocalizedObject('Tapahtuman lyhytkuvaus 3'),
+    pEvent: fakePEvent({
+      nextOccurrence: fakeOccurrences(1, [
+        fakeOccurrence({
+          startTime: new Date(2020, 10, 22, 12, 40).toISOString(),
+        }),
+      ]),
+    }),
+    keywords: fakeKeywords,
+  },
+  {
+    name: fakeLocalizedObject('Testitapahtuma 4'),
+    shortDescription: fakeLocalizedObject('Tapahtuman lyhytkuvaus 4'),
+    pEvent: fakePEvent({
+      nextOccurrence: null,
+      occurrences: fakeOccurrences(0),
+    }),
+    keywords: fakeKeywords,
+  },
+];
+
+const mocks: MockedResponse[] = [
+  {
+    request: {
+      query: EventsDocument,
+      variables: {
+        include: ['keywords', 'location'],
+        keyword: [],
+        allOngoingAnd: null,
+        inLanguage: '',
+        location: '',
+        start: 'now',
+        pageSize: 10,
+        sort: 'start_time',
+        end: null,
+        organisationId: '',
+        division: '',
+      },
+    },
+    result: {
+      data: {
+        events: fakeEvents(4, eventMocks),
+      },
+    },
+  },
+];
+
+afterEach(() => {
+  clear();
+});
+
+test('renders search form and events list with correct information', async () => {
```

Comment:
I think the focusing should be tested, because it is quite important in sense of accessibility. Another thing that could have tests is the expanded search when there are some filters active.

Reply:
I manually tested the focusing and tried the voice over also. The voice over speaks meaningful things, but I'm not sure if it's enough :)